### PR TITLE
[Android] Remote

### PR DESCRIPTION
**Pull request by talregev**
Fix: #1699 #4233  #4231
Add Category Remote.
Add check box remoteEable.
If check box enable, then show key.
Monitor:
Add quitClient.

1_Screenshot_20210302-202555_BOINC
2_Screenshot_20210302-202612_BOINC

### CODE REVIEW

**Comment by talregev** [comment]
* [x]  Verify that client build on Android is able to run with `--allow_remote_gui_rpc  `
* [x]  Add support of tcp/ip connection from Android BOINC Manager
* [x]  Add implementation of 'Enable Remote Monitoring checkbox' (from #4230)
* [x]  If user confirms remote monitoring, client should be restarted with command line option `--allow_remote_gui_rpc` without `--gui_rpc_unix_domain`
* [x]  If user unchecks remote monitoring checkbox, they should receive confirmation dialog that client will be relaunched and all running tasks will be interrupted
* [x]  If user confirm disabling remote monitoring, client should be relaunched with command line option `--gui_rpc_unix_domain` without `--allow_remote_gui_rpc`
* [x]  Restart client when password changed
* [x]  Add checkbox in Advanced settings 'Enable Remote Monitoring'
* [x]  Add description under checkbox: 'When changed, BOINC client will be relaunched.'
* [x]  Checkbox should be disabled by default

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.kt [hunk 1]
```diff
@@ -118,3 +124,24 @@ inline fun Long.secondsToLocalDateTime(
 
 @Suppress("NOTHING_TO_INLINE")
 inline fun Context.getColorCompat(@ColorRes colorId: Int) = ContextCompat.getColor(this, colorId)
+
+class TaskRunner {
+    private val executor: Executor = Executors.newSingleThreadExecutor() // change according to your requirements
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.kt:129]: Please remove that comment

File: android/BOINC/app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -129,6 +129,7 @@
     <string name="prefs_category_memory">Memory</string>
     <string name="prefs_category_other">Other</string>
     <string name="prefs_category_debug">Debug</string>
+    <string name="prefs_category_remote">Remote</string>
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/res/values/strings.xml:132]: `Remote` -> `Remote Control`

File: android/BOINC/app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -159,6 +160,8 @@
     <string name="prefs_network_daily_xfer_limit_period_days_header">Daily transfer limit (Days)</string>
     <string name="prefs_network_wifi_only_header">Transfer tasks on WiFi only</string>
     <string name="prefs_autostart_header">Autostart</string>
+    <string name="prefs_remote_header">Enable Remote</string>
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/res/values/strings.xml:163]: `Enable Remote` -> `Enable Remote Control`

**Review by AenBleidd — CHANGES_REQUESTED** [review]

File: android/BOINC/app/src/main/res/values/strings.xml (hunk 3 above)
Comment (talregev) [reply android/BOINC/app/src/main/res/values/strings.xml]: Fix

File: android/BOINC/app/src/main/res/values/strings.xml (hunk 2 above)
Comment (talregev) [reply android/BOINC/app/src/main/res/values/strings.xml]: Fix

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.kt (hunk 1 above)
Comment (talregev) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.kt]: Fix

**Review by AenBleidd — APPROVED** [review]

**Pull request merged**